### ISSUE TITLE
🐛 BUG: Nesting inside `<p>` tags

### ISSUE THREAD

**Issue by VodnikAngus** [withastro/compiler#432]
### What version of `@astrojs/compiler` are you using?

1.0.0-beta.47

### What package manager are you using?

pnpm

### What operating system are you using?

Fedora Linux

### Describe the Bug

Currently compiler doesn't allow nesting `<table>` and `<p>` elements inside other `<p>` elements as per HTML standard. But instead of showing a warning or error it just renders unexpected html:

```jsx
<p>
    <table />
</p>
```
gets rendered as:
```jsx
<p></p>
<table />
<p></p>
```

### Link to Minimal Reproducible Example

N / A

**Comment by lostra01** [comment]
This is the excepted behaviour. P tags get auto closed and they are added as siblings. Any online HTML Editor as well as web servers do the same, and will output the same.

**Comment by Mahmoudgalalz** [comment]
Is there's anyone work on this issue, otherwise could I take it, 
a little help, where I can find in the Tokenizes the html tags ?

**Comment by MoustaphaDev** [comment]
When trying to reproduce this, this was the output I got:
```html
<p>
    <table></table>
</p>
```
I believe this is the expected output. We don't show warnings/errors for invalid markup. If invalid, the compiler tries to correct the markup following the html standard.

Pinging @natemoo-re for a gut check

**Comment by Princesseuh** [comment]
Hello, the Go compiler for Astro is no longer maintained. All future development will be focused on [our new Rust compiler](https://github.com/withastro/compiler-rs). If this issue is still relevant and you can reproduce it on the Rust compiler, please open a new issue there. Thank you!
